sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2954,6 +2954,15 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val LEGACY_CHAR_VARCHAR_AS_STRING =
+    buildConf("spark.sql.legacy.charVarcharAsString")
+      .internal()
+      .doc("When true, the parser will not fail to parse char and varchar type but treat it as" +
```

Comment:
It seems this config is not related only to the parser because `failIfHasCharVarchar` is used in many places. Could you update it accordingly?

Reply:
seems that they are all related to the parser and actually char/varchar shouldn't exist after the parser phase

Reply:
Ah, ok. If so, how about `spark.sql.` -> `spark.sql.parser`?

Reply:
for example, `spark.read.schema(StructType)`  has no parser involved.